python/ray/serve/router.py

```diff
@@ -316,13 +316,14 @@ async def _do_query(self, backend, backend_replica_tag, req):
         start = time.time()
         worker = self.replicas[backend_replica_tag]
         try:
+            object_ref = worker.handle_request.remote(req.ray_serialize())
             if req.is_shadow_query:
                 # No need to actually get the result, but we do need to wait
                 # until the call completes to mark the worker idle.
-                asyncio.wait([worker.handle_request.remote(req)])
+                await asyncio.wait([object_ref])
```

Comment:
This is a bug before, we should have waited for the object

Reply:
Oh wow, good catch!